### PR TITLE
146 display password requirements

### PR DESCRIPTION
**Pull request by muratcan23**

### CODE REVIEW

File: src/pages/RegisterAndLogin/Components/Inputs.tsx [hunk 1]
```diff
@@ -134,11 +135,7 @@ const Inputs = ({ page }: { page: string }) => {
               children={open ? <VisibleIcon /> : <InvisibleIcon />}
             />
           </InputGroup>
-          {formErrors.password ? (
-            <Text color="red" as="i">
-              {formErrors.password}
-            </Text>
-          ) : null}
+          <PassRqrmt formData={formData} />
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/Inputs.tsx:138]: You should use better naming :)

File: src/pages/RegisterAndLogin/Components/PassRqrmt.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    <Flex
+      flexDirection="column"
+      w="350px"
+      h="185px"
+      boxShadow="xs"
+      p="1px"
+      rounded="md"
+      bg="white"
+    >
+      <Flex flexDirection="row" alignItems="flex-start">
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/PassRqrmt.tsx:29]: The default direction value is already row. No need to add it. And also there is no need to align items. I removed it nothing changed. Think using Hstack

File: src/pages/RegisterAndLogin/utils/InputValidation.ts [hunk 3]
```diff
@@ -3,7 +3,6 @@ export const InputValidation = (name: string, value: string) => {
     email: "",
     password: ""
   };
-
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/utils/InputValidation.ts:6]: This change should not be here

File: src/pages/RegisterAndLogin/Components/Inputs.tsx (hunk 1 above)
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/Inputs.tsx:138]: We have one big mistake. No need to show this section during login. It should be visible during registration.

File: src/pages/RegisterAndLogin/Components/PassRqrmt.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        >
+          Required!
+        </Text>
+      </Flex>
+      <Flex>
+        <Icon
+          as={RedTickIcon}
+          color={hasPsassword7chars ? "green" : "#e53e3e"}
+        />
+        <Text
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/PassRqrmt.tsx:44]: This part is repeated. Try to think to create a component and use it everywhere.
```
 <Text
          fontSize="sm"
          fontFamily="unset"
          color={hasPsassword7chars ? "green" : "#e53e3e"}
        >
          At least seven characters
        </Text>
```

File: src/pages/RegisterAndLogin/Components/PassRqrmt.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          fontSize="sm"
+          fontFamily="unset"
+          color={hasPsassword7chars ? "green" : "#e53e3e"}
+        >
+          At least seven characters
+        </Text>
+      </Flex>
+      <Flex>
+        <Icon as={RedTickIcon} color={hasPsNumber ? "green" : "#e53e3e"} />
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/PassRqrmt.tsx:53]: The icon should be changed depending on the condition as we discussed

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/pages/RegisterAndLogin/Components/PassRqrmt.tsx (hunk 2 above)
Comment (muratcan23) [reply src/pages/RegisterAndLogin/Components/PassRqrmt.tsx:29]: I tried Hstack and the first line broke the align then I backed to Flex and removed unnecessary flexDirectio and alignItems.
Screenshot 2023-03-13 230312

File: src/pages/RegisterAndLogin/Components/Inputs.tsx [hunk 6]
```diff
@@ -20,11 +20,13 @@ import {
 import { PATHS } from "../../../utils/paths";
 import { API } from "../../../utils/usedApi";
 import { InputValidation } from "../utils/InputValidation";
+import PassStandards from "./PassStandards";
 
 const Inputs = ({ page }: { page: string }) => {
   const toast = useToast();
   const [disabled, setDisabled] = React.useState(true);
   const navigate = useNavigate();
+  const [passStandard, setPassStandard] = React.useState(page === "register");
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/Inputs.tsx:29]: Do we really need a state here?

File: src/pages/RegisterAndLogin/Components/Inputs.tsx [hunk 7]
```diff
@@ -61,6 +63,7 @@ const Inputs = ({ page }: { page: string }) => {
 
   const submitHandler = () => {
     if (page === "register") {
+      setPassStandard(true);
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/Inputs.tsx:66]: I dont think we need state or setState

File: src/pages/RegisterAndLogin/Components/Inputs.tsx [hunk 8]
```diff
@@ -134,11 +137,7 @@ const Inputs = ({ page }: { page: string }) => {
               children={open ? <VisibleIcon /> : <InvisibleIcon />}
             />
           </InputGroup>
-          {formErrors.password ? (
-            <Text color="red" as="i">
-              {formErrors.password}
-            </Text>
-          ) : null}
+          {passStandard && <PassStandards formData={formData} />}
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/Inputs.tsx:140]: create a variable like this `const isRegisterPage= page === "register"` and use is everyhwhere that we used _page === "register"_ .

Here basically you can do it 
`{isRegisterPage&& <PasswordRequirements formData={formData} />}`

File: src/pages/RegisterAndLogin/Components/PassStandards.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+          color={hasPsNumber ? "green" : "#E53E3E"}
+        />
+        <Text
+          ml="5px"
+          fontSize="sm"
+          fontFamily="unset"
+          color={hasPsNumber ? "green" : "#e53e3e"}
+        >
+          At least one number
+        </Text>
```
Comment (kod-man) [inline src/pages/RegisterAndLogin/Components/PassStandards.tsx:70]: This part can be a card component since we are repeating it.

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/pages/RegisterAndLogin/Components/PassStandards.tsx (hunk 9 above)
Comment (muratcan23) [reply src/pages/RegisterAndLogin/Components/PassStandards.tsx]: I did it. I have a suggestion about the name: I gave the name "PassSpecsCard" to a new component, should I change "PassStandards"  to => "PassSpecs"

**Review by kod-man — APPROVED** [review]

**Pull request merged**